video/decode/vd_lavc.c

```diff
@@ -583,8 +585,13 @@ static void init_avctx(struct dec_video *vd, const char *decoder,
     AVCodecContext *avctx = ctx->avctx;
     if (!ctx->avctx)
         goto error;
+    avctx->opaque = vd;
     avctx->codec_type = AVMEDIA_TYPE_VIDEO;
     avctx->codec_id = lavc_codec->id;
+    if ((lavc_codec->capabilities & AV_CODEC_CAP_DR1) && vd->vo->driver->get_buffer) {
+        avctx->get_buffer2 = get_buffer2_direct;
+        avctx->thread_safe_callbacks = vd->vo->driver->caps & VO_CAP_GET_BUFFER_THREADSAFE;
```

Comment:
That would write the value of VO_CAP_GET_BUFFER_THREADSAFE, not 0 or 1. I don't know if that matters for libavcodec, but still might be a good idea not to write random values.

Also, I think we should always require this. Since get_buffer2_direct doesn't lock anything, and the decoder is on a different thread than the VO.